apps/google-analytics-4/frontend/src/components/InstalledServiceAccountKey.tsx

```diff
@@ -34,6 +36,31 @@ interface InstalledServiceAccountKeyProps {
 }
 
 const InstalledServiceAccountKey = ({ serviceAccountKeyId }: InstalledServiceAccountKeyProps) => {
+  const [credentialsData, setCredentialsData] = useState<Credentials | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState('');
+
+  useEffect(() => {
+    (async () => {
+      const api = new Api();
```

Comment:
Every time we make an api request, we have to create a new instance of `Api()`. `Api` can be initialized without any arguments. Should this be a singleton?

Reply:
Yes it's a good question / call out. In truth we can just instantiate it in the config and use it directly. I can always move it out later if there was a need to initialize it with values.

(e.g. I was thinking about if you wanted to pass an AbortController to cancel the request, it might make better sense to initialize the client with the abort controller vs. to pass it down to every Api method we implement.)